File: LeetCode_Algorithm/257(Binary Tree Paths)/257.py
from typing import Optional
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> list[str]:
        # 34ms 73% 14MB 23%
        self.ans = []
        self.solve(root, "")
        return self.ans
    def solve(self, node, temp):
        if not node:
            return ;
        temp = temp + f"{node.val}"
        if not node.left and not node.right:
            self.ans.append(temp)
            return ;
        self.solve(node.left, temp + "->")
        self.solve(node.right, temp + "->")

        

File: LeetCode_Algorithm/257(Binary Tree Paths)/257.js
var solve = (node, temp) => {
  if (!node) {
    return ;
  }
  if (!node.left && !node.right) {
    temp += String(node.val)
    ans.push(temp)
    return ;
  }
  temp += (String(node.val) + "->")
  solve(node.left, temp)
  solve(node.right, temp)
}
var binaryTreePaths = function(root) {
  // sol1 77ms 36% 43.5MB 70%
  ans = []
  solve(root, "")
  return ans
};